processing/src/main/java/io/druid/query/TableDataSource.java

```diff
@@ -21,13 +21,29 @@
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.google.common.base.Function;
+import com.google.common.collect.Lists;
 
 import java.util.Arrays;
 import java.util.List;
 
 @JsonTypeName("table")
 public class TableDataSource implements DataSource
 {
+  public static List<TableDataSource> of(List<String> dataSources)
+  {
+    return Lists.transform(
```

Comment:
Please wrap in `ImmutableList.copyOf`, to avoid continuous recomputation & allocation.